src/libcore/iter/mod.rs

```diff
@@ -520,6 +520,35 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
 #[unstable(feature = "fused", issue = "35602")]
 impl<I> FusedIterator for Cycle<I> where I: Clone + Iterator {}
 
+/// An iterator that skips n elements every iteration.
+///
+/// This `struct` is created by the [`every_nth`] method on [`Iterator`]. See
+/// its documentation for more.
+///
+/// [`every_nth`]: trait.Iterator.html#method.every_nth
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[derive(Clone, Debug)]
+pub struct EveryNth<I> {
+    iter: I,
+    step: usize,
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> Iterator for EveryNth<I> where I: Iterator {
+    type Item = I::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let elt = self.iter.next();
+        if self.step > 0 {
+            self.iter.nth(self.step - 1);
```

Comment:
Even if the fused/`first_take` approaches have the same branching, the current implementation and the one I pointed out do not have the same complexity from the point of view of work on the inner iterator, because of the early `nth`.
If the final consumer of the `every_nth` iterator only takes the first element, the underlying iterator will consume 1+n elements with the current implementation (as in @nagisa's first option), while one might expect it to only consume 1 element (as in the second option).